app/schema/schema.go

```diff
@@ -93,3 +113,32 @@ func Action_SchemaParse(args *cli.Context) error {
 	// TODO: or do something else if the "save" flag is set.
 	return ipld.EncodeStreaming(args.App.Writer, dmtn, encoder)
 }
+
+func Action_GoCodegen(args *cli.Context) error {
+	if args.NArg() != 1 {
+		return fmt.Errorf("invalid number of arguments")
+	}
+
+	s, err := schemadsl.ParseFile(args.Args().First())
+	if err != nil {
+		return err
+	}
+
+	var ts schema.TypeSystem
+	ts.Init()
+	if err := schemadmt.Compile(&ts, s); err != nil {
+		return err
+	}
+
+	outputDir := args.Path("output")
+	pkgName := args.String("package")
+
+	if err := os.MkdirAll(outputDir, os.ModePerm); err != nil {
+		return err
+	}
+	a := gengo.AdjunctCfg{}
+	gengo.Generate(fmt.Sprintf("%s", outputDir), pkgName, ts, &a)
+
+	// TODO: Add support for generating bindnode types once they're exported
```

Comment:
FYI, generating both at the same time doesn't seem particularly useful - I think most users will want one or the other.